include/pdn.h

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ *
+ * @brief Interface for PDN connection controlling.
+ *        No data can be exchanged over a PDN interface.
+ *        The interface is used to manage connections to specific APNs.
+ */
+
+/**
+ * @file pdn.h
+ *
+ * @brief Public APIs for the PDN library.
+ */
+
+#ifndef PDN_H_
+#define PDN_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief State of the PDN
+ */
+enum pdn_state { PDN_DOWN = 0, PDN_UP = 1 };
+
+/** @brief Notify event types
+ */
+enum pdn_event {
+	PDN_EVENT_NA = 0,
+	/* TODO: TBD */
+};
+
+/**
+ * @typedefs pdn_notif_handler_t
+ *
+ * This notification handler let's application to receive notification
+ * about incoming PDN events. Notifications will be dispatched to handler
+ * registered using the @ref pdn_register_handler() function. Handlers
+ * can be de-registered using the @ref pdn_deregister_handler() function.
+ *
+ * @param[in] handle    Pointer to pdn handle.
+ * @param[in] event     Event id @ref enum pdn_event.
+ */
+typedef void (*pdn_notif_handler_t)(void *handle, const enum pdn_event event);
+
+/**
+ * @brief Routine to open a PDN channel.
+ *
+ * @return  New handle if the procedure succeeds else NULL.
+ *          errno indicates the reason for failure.
+ */
+void *pdn_open(void);
+
+/**
+ * @brief Routine to close a PDN socket.
+ *
+ * @param[in] handle    Handle returned by @ref pdn_open().
+ *
+ * @return  0 if the procedure succeeds, else, -1.
+ *          errno indicates the reason for failure.
+ */
+int pdn_close(void **handle);
+
+/**
+ * @brief Function to connect a PDN socket to a remote address.
+ *
+ * @param[in] handle           Handle returned by @ref pdn_open().
+ * @param[in] p_remote_addr    Pointer to a structure containing the remote
```

Comment:
Looking at how it's used in the sample and implemented,  it's just a `char *` not a pointer to a structure.  Why not make it `const char *` instead of enigmatic `const void *`?

Also, drop the `p_` prefix from the pointer, we don't use Hungarian notation in NCS.

Reply:
I already changed the interface and this is going to be changed to accordingly.